vortex-sampling-compressor/src/sampling_compressor.rs

```diff
@@ -179,8 +183,9 @@ impl<'a> SamplingCompressor<'a> {
             return cc.compress(array, None, self.clone());
         }
 
-        if let Some(cc) = ConstantCompressor.can_compress(array) {
-            return cc.compress(array, None, self.clone());
+        // short-circuit because seriously nothing beats constant
+        if self.is_enabled(&ConstantCompressor) && ConstantCompressor.can_compress(array).is_some() {
```

Comment:
Was the brokeness here that we disabled the compressor and looped forever?

Reply:
no, it was that we disabled ConstantCompressor when compressing FSST codes (since we only want to compress the offsets inside the codes VarBin), but this ignored that. FSST::try_new requires codes to have VarBin encoding